titus-common/src/main/java/com/netflix/titus/common/util/FunctionExt.java

```diff
@@ -37,4 +40,14 @@
         }
         return opt;
     }
+
+    /**
+     * {@link UnaryOperator} does not have an equivalent of {@link Function#andThen(Function)}, so this can be used when
+     * the more specific type is required.
+     */
+    public static <T> UnaryOperator<T> andThen(UnaryOperator<T> op, UnaryOperator<T> after) {
```

Comment:
By `more specific type` here I mean returning back a `UnaryOperator` rather than a `Function` (I’m not referring to the generic type `T`). The implementation is just a copy-and-pasta of `Function#andThen(Function)`, but returning a `UnaryOperator` (more specifically type) back.

Does it mane more sense now? Any suggestion on how to make the doc more clear?

Reply:
Makes sense. Maybe you can change it to: `the more specific type (UnaryOperator) is required` to avoid any confusion.